BezierCurve.cpp

```diff
@@ -0,0 +1,150 @@
+#include "BezierCurve.h"
+
+BezierCurve::BezierCurve(coord a, coord b):
+    mSize(2),
+    mCoords{a, b, {}}
+{
+}
+
+BezierCurve::BezierCurve(coord a, coord mid1, coord b):
+    mSize(3),
+    mCoords{a, mid1, b}
+{
+}
+
+template<typename T>
+inline T abs(T a)
+{
+    if (a < 0)
+    {
+        return -a;
+    }
+    return a;
+}
+
+template<typename T>
+inline void swap(T& a, T& b)
+{
+    T c(a);
+    a = b;
+    b = c;
+}
```

Comment:
These probably need to be moved to a separate file, but haven't made it there yet.